### ISSUE TITLE
Support magic comment for ignoring specific blocks

### ISSUE THREAD

**Issue by mcandre** [RedCoolBeans/dockerlint#7]
E.g., I need an `ADD` in order to copy and extract a stage3 tarball for building a Gentoo container. Docker doesn't yet have an "EXEC" command to accomplish this. I want to lint my Dockerfile, but ignore the necessary `ADD`.

**Comment by jasperla** [comment]
You're looking for something like the Lint style comments like `/* NOTREACHED */` , right?

**Comment by mcandre** [comment]
yeah.

**Comment by jasperla** [comment]
Actually, I wonder why Dockerlint is warning as the check has this comment:

```
# Although ADD and COPY are functionally similar, generally speaking, COPY is preferred.
# The only/best use for ADD is to add-and-extract archives to an image.
# Reports: WARN
```

So I wonder why it's complaining for you when you use `ADD` with a tarball. Could you please show the offending line from your Dockerfile?

**Comment by mcandre** [comment]
https://github.com/mcandre/docker-gentoo/blob/master/Dockerfile#L4

**Comment by jasperla** [comment]
I don't see any problems in linting your `Dockerfile`:

```
DEBUG:[ { line: 1, instruction: 'FROM', arguments: [ 'scratch' ] },
  { line: 2,
    instruction: 'MAINTAINER',
    arguments: [ 'Andrew Pennebaker <andrew.pennebaker@gmail.com>' ] },
  { line: 3, instruction: 'COPY', arguments: [ 'tar /tar' ] },
  { line: 4,
    instruction: 'ADD',
    arguments: [ 'stage3-amd64-20150604.tar.bz2 /' ] },
  { line: 5,
    instruction: 'RUN',
    arguments:
     [ 'echo \'default\' > /run/openrc/softlevel &&',
       'sed -e \'s/#rc_sys=""/rc_sys="lxc"/g\' -i /etc/rc.conf &&',
       'ln -s /etc/init.d/net.lo /run/openrc/started/net.lo &&',
       'ln -s /etc/init.d/net.lo /etc/init.d/net.eth0 &&',
       'ln -s /etc/init.d/net.eth0 /run/openrc/started/net.eth0 &&',
       'echo \'UTC\' > /etc/timezone &&',
       'mkdir -p /usr/portage/{distfiles,metadata,packages} &&',
       'chown -R portage:portage /usr/portage &&',
       'echo \'masters = gentoo\' > /usr/portage/metadata/layout.conf &&',
       'emerge-webrsync &&',
       'eselect news read new &&',
       'env-update' ] } ]

INFO: Dockerfile.mcandre is OK.
```

There is a specific check in the code to allow `ADD` for archives, such as the `.tar.bz2` extension.

**Comment by DaleJefferson** [comment]
I seem to have a false positive as well.

```
ENV INSTALL_DIR /opt/foo/bar

WORKDIR $INSTALL_DIR // WORKDIR path $INSTALL_DIR must be absolute
```

An ignore comment would be really useful.

**Comment by benjamine** [comment]
+1

**Comment by rnagy** [comment]
@DaleJefferson that has been fixed in 1fa5bfe3e79bdc4a7e43ab2980fa116d003fb5d1

**Comment by rnagy** [comment]
Closing the initial ticket as the point of linting is not to have a way around issues.

**Comment by dmikalova** [comment]
This would be nice to have because of [issue 9](https://github.com/RedCoolBeans/dockerlint/issues/9).

**Comment by shellscape** [comment]
> Closing the initial ticket as the point of linting is not to have a way around issues.

Holy bananas that's a bit of a strong position to take. I'd assert that things have changed since 2017, and that all major linters (ESLint, etc) support disabling rules, lines, blocks, etc. Sometimes you _want to willingly_ disable the linter for any number of reasons.

Linters are supposed to be opinionated by default, but configurable.
